Assets/Tests/PlayModeTests/UserProfileTests.cs

```diff
@@ -449,6 +449,312 @@ public void SessionTimerElapseWithRecordedValue_M_CR_CG()
             TestUtility.ValidateUserDetails(up1, ExpectedUserProperty());
             cly.RequestHelper._requestRepo.Clear();
         }
+        
+        // Initializes Countly and returns the Countly instance with clear repository
+        private Countly BaseInit()
+        {
+            Countly cly = Countly.Instance;
+            CountlyConfiguration config = TestUtility.CreateBaseConfig();
+            cly.Init(config);
+
+            // session request
+            TestUtility.ValidateRQEQSize(cly, 1, 0);
+            cly.RequestHelper._requestRepo.Clear();
+
+            return cly;
+        }
+
+        // 'Increment' in Countly.Instance.UserProfile
+        // We initialize the sdk with no consent requirement, try to record user profile data with 'Increment' and call save
+        // Since we call 'Increment' with bad keys, it should not record any user profile data
+        [Test]
+        public void Increment_BadKey_CNR()
+        {
+            Countly cly = BaseInit();
+            
+            cly.UserProfile.Increment(null);
+            cly.UserProfile.Save();
+            TestUtility.ValidateRQEQSize(cly, 0, 0);
+
+            cly.UserProfile.Increment(" ");
+            cly.UserProfile.Save();
+            TestUtility.ValidateRQEQSize(cly, 0, 0);
+
+            cly.UserProfile.Increment("");
+            cly.UserProfile.Save();
+            TestUtility.ValidateRQEQSize(cly, 0, 0);
+        }
+        
+        // 'IncrementBy' in Countly.Instance.UserProfile
+        // We initialize the sdk with no consent requirement, try to record user profile data with 'IncrementBy' and call save
+        // Since we call 'IncrementBy' with bad keys, it should not record any user profile data
+        [Test]
+        public void IncrementBy_BadKey_CNR()
+        {
+            Countly cly = BaseInit();
+
+            cly.UserProfile.IncrementBy(null, 10);
+            cly.UserProfile.Save();
+            TestUtility.ValidateRQEQSize(cly, 0, 0);
+
+            cly.UserProfile.IncrementBy(" ", 10);

```

Comment:
ow whitespace is not valid?

Reply:
whitespace and null values are ignored btw